sts-secret-server/src/test/java/de/adorsys/sts/secretserver/CachingKeyRetrieverServiceThrowsTest.java

```diff
@@ -0,0 +1,30 @@
+package de.adorsys.sts.secretserver;
+
+import com.nimbusds.jose.jwk.JWKSet;
+import de.adorsys.sts.resourceserver.service.CachingKeyRetrieverService;
+import de.adorsys.sts.resourceserver.service.KeyRetrieverService;
+import org.junit.Test;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class CachingKeyRetrieverServiceThrowsTest {
+    private KeyRetrieverService retrieverService = mock(KeyRetrieverService.class);
+    @Test
+    public void testThrows() {
```

Comment:
It is better to test the following cases:
1. No exception thrown -> cache
1. Exception -> cache, read cache, then no exception -> cache
1. Exception-> cache, read cache